docs/docs/terraform/main.tf

```diff
@@ -0,0 +1,466 @@
+  protocol    = "TCP"
+  target_type = "ip"
+  vpc_id = data.aws_vpc.neptune_contained_herein.id
+  deregistration_delay = 0
+}
+
+resource "aws_lb_target_group_attachment" "service" {
+  count = length(aws_neptune_cluster.service)
+  target_group_arn = aws_lb_target_group.service[count.index].arn
+  # for aws_lb_target_group with target_type = "ip", target_id is actually an IP address
+  target_id = data.dns_a_record_set.aws_neptune_cluster_service[count.index].addrs[0]
+  port = aws_neptune_cluster.service[count.index].port
+}
+
+resource "aws_lb_listener" "service" {
+  count = length(aws_neptune_cluster.service)
+  load_balancer_arn = aws_lb.service[count.index].arn
+  port = aws_neptune_cluster.service[count.index].port
+  protocol = "TCP"
+  default_action {
+    type             = "forward"
+    target_group_arn = aws_lb_target_group.service[count.index].arn
+  }
+}
+
+resource "aws_s3_bucket" "bulk_loader" {
+  bucket = "${var.service}-${var.environment}-bulk-loader"
+  # acl="private" is not the same as the aws_s3_bucket_public_access_block
+
+  lifecycle_rule {
+    abort_incomplete_multipart_upload_days = 1
+    enabled                                = true
+    id                                     = "delete old"
+    tags                                   = {}
+    expiration {
+      days                         = 2
+      expired_object_delete_marker = false
+    }
+
+    noncurrent_version_expiration {
+      days = 1
+    }
+  }
+}
+
+resource "aws_s3_bucket_public_access_block" "bulk_loader" {
+  bucket = aws_s3_bucket.bulk_loader.id
+  block_public_acls   = true
+  block_public_policy = true
+  ignore_public_acls  = true
+  restrict_public_buckets = true
+}
+
+resource "aws_vpc_endpoint" "s3" {
+  vpc_id       = aws_default_vpc.default.id
+  service_name = "com.amazonaws.${data.aws_region.current.name}.s3"
+}
+
+# iam things
+resource "aws_iam_role" "neptune_load_from_s3" {
+  name = "NeptuneLoadFromS3"
+  description = "Allows Neptune to call S3 services on your behalf."
+
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "Service": "rds.amazonaws.com"
+      },
+      "Action": "sts:AssumeRole"
+    }
+  ]
+}
+EOF
+}
+
+locals {
+  neptune_load_from_s3_policies = [
+    "arn:aws:iam::aws:policy/AmazonS3ReadOnlyAccess",
+  ]
+}
+
+resource "aws_iam_role_policy_attachment" "neptune_load_from_s3" {
+  role       = aws_iam_role.neptune_load_from_s3.name
+  policy_arn = local.neptune_load_from_s3_policies[count.index]
+  count = length(local.neptune_load_from_s3_policies)
+}
+
+data "aws_region" "current" {}
+
+resource "aws_iam_user" "service" {
+  name = var.service
+}
+
+resource "aws_iam_user" "dataloader" {
+  name = "dataloader"
+}
+
+locals {
+  # put the GPG public key you wish to encrypt the secrets with here
+  # gpg1 --no-default-keyring --keyring ./whatever-key-ring --export whatever-key-id  | base64
+  pgp_key = "...."
```

Comment:
Should this be called out in the `neptune_for_amundsen.md` doc someplace?

Reply:
Hmmm, maybe? It feels like the neptune for amundsen doc is more general guidance, whereas this terraform is more of an example you can use or model your impl after (you don't need to use a pgp_key here if you would rather not)

Reply:
Seems reasonable to leave as-is 👍